internal/handlers/shared/msg_listdatabases.go

```diff
@@ -25,53 +24,92 @@ import (
 )
 
 func (h *Handler) MsgListDatabases(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	var names []string
-	rows, err := h.pgPool.Query(ctx, "SELECT schema_name FROM information_schema.schemata")
+	document, err := msg.Document()
 	if err != nil {
-		return nil, err
+		return nil, lazyerrors.Error(err)
 	}
-	defer rows.Close()
 
-	for rows.Next() {
-		var name string
-		if err = rows.Scan(&name); err != nil {
+	m := document.Map()
+
+	var databaseNames []string
+
+	db, ok := m["$db"].(string)
```

Comment:
I am checking if there is a db passed when querying listDatabases... Not sure if this was correct or not... I think there's an optional filter > db query that can be passed? So maybe this is useless

Reply:
Gotcha I was confused by this: 

`{
  "Checksum": 0,
  "FlagBits": [],
  "Sections": [
    {
      "Document": {
        "$k": [
          "listDatabases",
          "$db",
          "$readPreference"
        ],
        "listDatabases": 1,
        "$db": "admin",
        "$readPreference": {
          "$k": [
            "mode"
          ],
          "mode": "primaryPreferred"
        }
      },
      "Kind": 0
    }
  ]
}`

I'll fix this later :)

Reply:
@mangodb-bot Can you have a look now? I feel like I'm still wrong...